### ISSUE TITLE
Previous Editions of the Schematron Standard no more freely available?

### ISSUE THREAD

**Issue by nkutsche** [Schematron/awesome-schematron#7]
At least the links at the [Readme.md](https://github.com/Schematron/awesome-schematron/blob/master/README.md) are broken/miss-leading:

> - [ISO Schematron 2nd Edition](http://standards.iso.org/ittf/PubliclyAvailableStandards/c055982_ISO_IEC_19757-3_2016.zip) - ISO/IEC 19757-3:2016 - Information technology - Document Schema Definition Languages (DSDL) - Part 3: Rule-based validation - Schematron. Freely available.
> - [ISO Schematron 1st Edition](http://standards.iso.org/ittf/PubliclyAvailableStandards/index.html) - ISO/IEC 19757-3:2006 - Information technology -- Document Schema Definition Language (DSDL) -- Part 3: Rule-based validation -- Schematron. Freely available.

The link to the 2nd edition is broken, the one to the 1st edition leads to a page where I can find only the parts 2 (RelaxNG), 4 (NVDL) and 11 (Schema association) of DSDL ("ISO/IEC 19757"). 

@rjelliffe: on your [blog post about the 3rd edition](https://schematron.com/document/455.html) you also use the same broken link to the 2nd edition.

Anyone an idea where I can find the 2nd edition? On my drive I found only the 1st edition. 🙁

**Comment by Hibou57** [comment]
It's the first ISO one, ISO/IEC 19757-3:2006, so an old one, still here is that one: https://web.archive.org/web/20090617131654/https://www.xfront.com/schematron/ISO-Schematron-Specification.pdf . I searched for the 2016 version, but could not find it. I could find the 2006 version indirectly by chance.

The issue with the 2025 version is not that much that it is not free as in free beer (and beer are rarely free, unless gifted by someone), it is that the price is high for an individual to read it. I get informed there is a book, https://xmlpress.net/2026/05/31/schematron-2025-a-language-for-validating-xml/ , which is more affordable (€35) still a bit expansive but really more affordable than the ISO document (€222). But I don’t know if it provides all of the details that are expected to be found in the ISO standard. If it's just the wording which differs, this is not an issue, as far as it contains the same informations.

**Comment by Hibou57** [comment]
Found an early draft of the second edition, here : https://web.archive.org/web/20100714072120/https://www.itscj.ipsj.or.jp/sc34/open/1419.pdf
Linked from here, after searching the relevant backup on WebArchive : https://web.archive.org/web/20160511013855/http://broadcast.oreilly.com/2010/04/public-draft-of-next-generatio.html . This is the link labelled “SC34 Website (PDF)”
Linked from here : https://xml.org/archive/lists/xml-dev/months/201605/messages/77

The traceability provided, is to show its origin.

This is **not the second edition, it's an early draft** of the second edition.

**Comment by Hibou57** [comment]
Why not have an alternative similar standard, with a copyright which explicitly disallows companies likes ISO and similar, to claim ownership on it while they were not the author?

**Comment by rjelliffe** [comment]
ISO allows standards from external bodies to be dual-branded as ISO/IEC
standards.  So one industry organization (ECMA, OASIS) looks after
development of the standard and then the (slower) ISO process takes it up.
The text is reconciled so that it follows ISO norms.  So then the same text
is available from the industry organization and also from ISO.

Schematron is different because it did not come from an industry standards
group, but from an individual (me).  So I initially agreed to let it become
an ISO standard on the belief that the standard would be made available for
free, which it was in the first edition.

To help comments, I also put out early drafts (pre-voting) of the second
edition.

However, subsequently ISO "clarified" its rules: because they considered
Schematron to be an in-house standard (i.e. not coming from some recognized
external body), the free PDF version of the first edition was taken down
when the second edition was published, and ISO lawyers contacted me
demanding that I take down the public draft second edition as well, which
of course I did. (The legal world of ISO only caters for institutions and
their representatives, not individuals.) I ran out of energy.  So there is
no free version of the final standards for the second and third edition.
And Schematron's minimalist specification meant that Schematron is probably
the most expensive, in dollars per page, of any ISO Standard: it is utter
madness. I bet there is not a single copy of the ISO standard ever sold.
AFAIR I have never seen the second and third editions.

However, the legal terms of the schemas is specifically intended to allow
distribution  and copying freely (free as in beer and licence): this is
also true of ISO DSDL and ISO SGML by the way.  So I believe the best way
forward is for the schemas to contain better documentation of the intent
and usage of the elements and attributes.

A move to another standards body would entail a complete re-write of the
text, unless the working group convinced ISO.

**Comment by Hibou57** [comment]
Thanks for the enlightenment

> because they considered Schematron to be an in-house standard (i.e. not coming from some recognized external body)

Is that legal? It looks strange, at least and looks like denial of justice. Well, on the other hand if you agreed to let it become an ISO standard, it may be different. I don’t know elsewhere, but in france, an author can never transfer her/his ownership, only some rights to make use or distribute, never the ownership. I guess it's different elsewhere.

> And Schematron's minimalist specification meant that Schematron is probably the most expensive, in dollars per page, of any ISO Standard

Not really, there is worse: ISO 3951 which is 15 pages long and cost $281 or €246 (landed on an old version of it haphazardly, searching for something else).

If I specify a language in the future (which may be planed), I will explicitly disallow the ISO to take ownership on it, in the copyright, so that it will be clear.

That said, do you know if this book coveys the same information as the standard: https://xmlpress.net/2026/05/31/schematron-2025-a-language-for-validating-xml/ ? It's really more affordable

This story is a pity, the idea of Schematron has always been the one looking the most natural to me (and others, I see) for a validation language. I had defined one of the like for personal use, with paths too (not XPath, simpler), but it was obviously not standard.

Don’t be too harassed, may be it will change in the future

**Comment by rjelliffe** [comment]
Each edition of ISO Schematron is a superset of the previous version, so
older books do not go out-of-date, they just grow as Schematron evolves.
There are three books: I think think they are each very good, and was a
reviewer for some drafts:

   - Eric van der Vlist wrote a good little book for O'Reilly "Short cuts"
   in 2007: it is a good introduction but very brief (for XPath 1). Here is a
   review: https://schematron.com/document/3882.html
   - Erik Siegel's also wrote book in 2022 (For Xpath 2), and updated it in
   2025 for the latest edition (for XPath 3). Here is a review:
   https://schematron.com/document/3882.htm

Basically, Schematron is very simple and designed to promote
well-structured XPaths: much of the power is in using XPaths well, most
complex Xpaths can be tamed using sch:let variables, and most complex
contexts with many conditions can be tamed using multiple sch:rules to
exclude cases. (Some users are surprised that an sch:pattern acts as a
switch or if-then operator, rather than all sch:rules in an sch:pattern
being tested independently. Schematron 2025 supplies an alternative to
sch:pattern with independent sch:rules, namely sch:rule-set.)

ISO does not control Schematron, they just have the copyright on the text
to a document which has been improved and vetted by lots of motivated and
smart people over the years so that it meets certain form and quality
constraints (always improvable), and which had enough international
interest to keep it a standard for 25 years.  So they are not villains or
bad guys: I am definitely a fan of most standards organizations, in
particular ISO who get so much right...it is just regrettable that their
model (made to allow standardized screws etc.) ultimately did not scale
down to a technology that was 100% grass-roots driven but failed to fit in
with the plans of the big players. Microsoft, for example, considered it
but told me that their problem was that if they took it on they would have
to support it for 20 years at least, which they could not project as a
revenue win: I gather it did not fit into their main use for XSD which was
"Strong typing", a euphemism for automated databinding rather than
validation.

It would be completely possible to produce an alternative specification for
Schematron, compatible with the public schemas and ISO semantics: there
might even be a way for development to shift to OASIS and take over
maintenance (the way that e.g. W3C MathML took over maintenance of the ISO
SGML standard entity set.) Or the community can define new QLBs (Query
Language Bindings) and new elements in a different namespace.  At one
stage, I had started adjusting Dave Pawson's free re-statement of
Schematron first edition for Schematron.com, but ran out of energy (I had
long ago decided that the way for Schematron to thrive was to get
completely out of the way and let community interest develop it: I have
seen cases where the founder of a technology stymies its evolution.)

If you have a need, get motivated and join the effort. And if you are
looking for a good Open Source project, there are lots of things that could
be done: for example, Java's validation API makes room for Schematron, and
the bad implementation in libxslt (2?) has finally bitten the dust, so
there are gaps where some standard implementation could go.

Rick

**Comment by tgraham-antenna** [comment]
> If you have a need, get motivated and join the effort.

For better or worse, an 'awesome-schematron' issue discussion isn't the best place to advance the effort or even to get much visibility. It might be better to take this to the Schematronist mailing list and see if this time around there's more of a groundswell of support for the effort.

**Comment by Hibou57** [comment]
> there are gaps where some standard implementation could go.

Yes, but how to design a standard implementation, without access to the standard? As I said before, I personally believe the issue is not that it is not free as in beer, rather that it too much expensive, especially when it is to work on a project for free as in beer.

**Comment by Hibou57** [comment]
> The issue with the 2025 version is not that much that it is not free as in free beer (and beer are rarely free, unless gifted by someone), it is that the price is high for an individual to read it. I get informed there is a book, https://xmlpress.net/2026/05/31/schematron-2025-a-language-for-validating-xml/ , which is more affordable (€35) still a bit expansive but really more affordable than the ISO document (€222). **But I don’t know if it provides all of the details that are expected to be found in the ISO standard.** If it's just the wording which differs, this is not an issue, as far as it contains the same informations.

Got the answer to this question, from the author her/him-self, after someone with good attention mentioned the same question elsewhere, to the author: https://github.com/Schematron/schema/issues/11#issuecomment-5355576524
